### PR TITLE
Add Workitem (UPS-RS) Feature [AB#87263]

### PR DESCRIPTION
**Pull request by shivakumar-ms**
## Description
Add Workitem Feature implementation

## Related issues
Addresses [AB#87263](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/87263).

## Testing
Unit Tests and Integration Tests.

### CODE REVIEW

File: src/Microsoft.Health.Dicom.Api/Controllers/WorkitemController.Add.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        [PartitionRoute(KnownRoutes.CreateWorkitemInstancesRoute, Name = KnownRouteNames.PartitionedWorkitemInstance)]
+        [VersionedRoute(KnownRoutes.CreateWorkitemInstancesRoute, Name = KnownRouteNames.VersionedWorkitemInstance)]
+        [Route(KnownRoutes.CreateWorkitemInstancesRoute, Name = KnownRouteNames.WorkitemInstance)]
+        [AuditEventType(AuditEventSubType.Workitem)]
+        public async Task<IActionResult> AddAsync(string workitemInstanceUid = null)
+        {
+            return await PostAsync(workitemInstanceUid);
+        }
+
+        // TODO: Add a POST call to accept instance UID from the header
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Api/Controllers/WorkitemController.Add.cs:75]: Remove this and extra new line

File: src/Microsoft.Health.Dicom.Api/Controllers/WorkitemController.Add.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+        private async Task<IActionResult> PostAsync(string workitemInstanceUid)
+        {
+            long fileSize = Request.ContentLength ?? 0;
+            _logger.LogInformation("DICOM Web Store Workitem Transaction request received, with UPS instance UID {UPSInstanceUid}, and file size of {FileSize} bytes",
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Api/Controllers/WorkitemController.Add.cs:80]: nit: Should we say WorkitemInstanceUID?

File: src/Microsoft.Health.Dicom.Api/Controllers/WorkitemController.Add.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                workitemInstanceUid ?? string.Empty,
+                fileSize);
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Api/Controllers/WorkitemController.Add.cs:82]: Do we need fileSize?

File: src/Microsoft.Health.Dicom.Api/Features/Routing/KnownActionParameterNames.cs [hunk 4]
```diff
@@ -12,6 +12,8 @@ internal class KnownActionParameterNames
         internal const string StudyInstanceUid = "studyInstanceUid";
         internal const string SeriesInstanceUid = "seriesInstanceUid";
         internal const string SopInstanceUid = "sopInstanceUid";
+        internal const string WorkItemInstanceUid = "workitemInstanceUid";
+        internal const string WorkItemTransactionUid = "transactionUid";
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Api/Features/Routing/KnownActionParameterNames.cs:16]: Where do we use this?

File: src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobWorkitemStore.cs [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+using Microsoft.Health.Dicom.Core.Features.Common;
+using Microsoft.Health.Dicom.Core.Features.Model;
+using Microsoft.Health.Dicom.Core.Web;
+using Microsoft.IO;
+using System.Text.Json;
+
+namespace Microsoft.Health.Dicom.Blob.Features.Storage
+{
+    /// <summary>
+    /// Provides functionality for managing the DICOM instance metadata.
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobWorkitemStore.cs:27]: Update comment

File: src/Microsoft.Health.Dicom.Core/Features/Workitem/IAddWorkitemResponseBuilder.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+// -------------------------------------------------------------------------------------------------
+
+using FellowOakDicom;
+using Microsoft.Health.Dicom.Core.Features.Store;
+using Microsoft.Health.Dicom.Core.Messages.WorkitemMessages;
+
+namespace Microsoft.Health.Dicom.Core.Features.Workitem
+{
+    /// <summary>
+    /// Provides functionality to build the response for the workitem store transaction.
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Features/Workitem/IAddWorkitemResponseBuilder.cs:13]: Update comment to `add workitem transaction`

File: src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemOrchestrator.cs [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+using Microsoft.Health.Dicom.Core.Features.Common;
+using Microsoft.Health.Dicom.Core.Features.Context;
+using Microsoft.Health.Dicom.Core.Features.Delete;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
+using Microsoft.Health.Dicom.Core.Features.Model;
+
+namespace Microsoft.Health.Dicom.Core.Features.Workitem
+{
+    /// <summary>
+    /// Provides functionality to orchestrate the storing of the DICOM instance entry.
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemOrchestrator.cs:21]: Update comment

File: src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemOrchestrator.cs [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+                int partitionKey = _contextAccessor.RequestContext.GetPartitionKey();
+
+                IReadOnlyCollection<QueryTag> queryTags = await _workitemQueryTagService.GetQueryTagsAsync(cancellationToken).ConfigureAwait(false);
+
+                long workitemKey = await _indexWorkitemStore
+                    .AddWorkitemAsync(partitionKey, dataset, queryTags, cancellationToken)
+                    .ConfigureAwait(false);
+
+                // TODO: this needs to be done before storing the dataset in the Blob Store.
+                string workitemInstanceUid = dataset.GetString(DicomTag.SOPInstanceUID);
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemOrchestrator.cs:63]: This is not valid anymore. Either we should get from `AffectedSOPInstanceUID` or we do this while retrieving

Comment (bcarthic) [reply src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemOrchestrator.cs:63]: Lets do in retrieve or update response builder

File: src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemOrchestrator.cs [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+                identifier = dataset.ToWorkitemInstanceIdentifier(workitemKey, partitionKey);
+
+                // We have successfully created the index, store the file.
+                await StoreWorkitemBlobAsync(identifier, dataset, cancellationToken)
+                    .ConfigureAwait(false);
+            }
+            catch
+            {
+                // TODO: implement cleanup to delete the index if blob storage fails
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemOrchestrator.cs:74]: Remove TODO

File: src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemRequestValidatorExtensions.cs [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+namespace Microsoft.Health.Dicom.Core.Features.Workitem
+{
+    internal static class WorkitemRequestValidatorExtensions
+    {
+        /// <summary>
+        /// Validates an <see cref="AddWorkitemRequest"/>.
+        /// </summary>
+        /// <param name="request">The request to validate.</param>
+        /// <exception cref="BadRequestException">Thrown when request body is missing.</exception>
+        /// <exception cref="UidValidation">Thrown when the specified StudyInstanceUID is not a valid identifier.</exception>
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemRequestValidatorExtensions.cs:20]: Update comment

File: src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemSerializer.cs [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+
+                IEnumerable<DicomDataset> datasets = JsonSerializer.Deserialize<IEnumerable<DicomDataset>>(json, serializerOptions);
+
+                return datasets;
+            }
+        }
+    }
+
+    /// <summary>
+    ///
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemSerializer.cs:50]: Update comment

File: src/Microsoft.Health.Dicom.Tests.Common/Comparers/DicomItemCollectionEqualityComparer.cs [hunk 12]
```diff
@@ -38,8 +38,8 @@ public bool Equals(IEnumerable<DicomItem> x, IEnumerable<DicomItem> y)
             IEqualityComparer<DicomItem> dicomItemComparer = new DicomItemEqualityComparer();
 
             ISet<DicomTag> ignoredSet = new HashSet<DicomTag>(_ignoredTags);
-            Dictionary<DicomTag, DicomItem> xDict = x.ToDictionary(item => item.Tag);
-            Dictionary<DicomTag, DicomItem> yDict = y.ToDictionary(item => item.Tag);
+            var xDict = x.ToDictionary(item => item.Tag);
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Tests.Common/Comparers/DicomItemCollectionEqualityComparer.cs:41]: We can undo this change

File: src/Microsoft.Health.Dicom.Core/Features/Workitem/AddWorkitemDatasetValidator.cs [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+            ValidateRequiredTags(dicomDataset, workitemInstanceUid);
+        }
+
+        private static void ValidateRequiredTags(DicomDataset dicomDataset, string workitemInstanceUid)
+        {
+            // The format of the identifiers will be validated by fo-dicom.
+            string workitemUid = EnsureRequiredTagIsPresent(dicomDataset, DicomTag.AffectedSOPInstanceUID);
+
+            // If the workitemInstanceUid is specified, then the workitemUid must match.
+            if (!string.IsNullOrWhiteSpace(workitemInstanceUid) &&
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Features/Workitem/AddWorkitemDatasetValidator.cs:34]: We also need to validate that the state is SCHEDULED and not other things

Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.Core/Features/Workitem/AddWorkitemDatasetValidator.cs:69]: Done

File: src/Microsoft.Health.Dicom.Tests.Common/Comparers/DicomItemCollectionEqualityComparer.cs (hunk 12 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.Tests.Common/Comparers/DicomItemCollectionEqualityComparer.cs]: Done

File: src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemSerializer.cs (hunk 11 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemSerializer.cs]: Done

File: src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemRequestValidatorExtensions.cs (hunk 10 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemRequestValidatorExtensions.cs]: Done

File: src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemOrchestrator.cs (hunk 9 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemOrchestrator.cs]: Done

File: src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemOrchestrator.cs (hunk 7 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemOrchestrator.cs]: Done

File: src/Microsoft.Health.Dicom.Core/Features/Workitem/IAddWorkitemResponseBuilder.cs (hunk 6 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.Core/Features/Workitem/IAddWorkitemResponseBuilder.cs]: Done.

File: src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobWorkitemStore.cs (hunk 5 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobWorkitemStore.cs]: Done

File: src/Microsoft.Health.Dicom.Api/Controllers/WorkitemController.Add.cs (hunk 1 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.Api/Controllers/WorkitemController.Add.cs]: Done

File: test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/DataPartitionEnabledHttpIntegrationTestFixture.cs [hunk 14]
```diff
@@ -15,7 +15,7 @@ public DataPartitionEnabledHttpIntegrationTestFixture()
         }
 
         protected DataPartitionEnabledHttpIntegrationTestFixture(string targetProjectParentDirectory)
-            : base(targetProjectParentDirectory, enableDataPartitions: true)
+            : base(targetProjectParentDirectory, enableDataPartitions: true, enableUpsRs: true)
```
Comment (bcarthic) [inline test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/DataPartitionEnabledHttpIntegrationTestFixture.cs:18]: We dont need to set here. This fixture is for testing features where feature flag wont turned on by default

File: src/Microsoft.Health.Dicom.Core/Extensions/WorkitemDatasetExtensions.cs [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+        /// <param name="workitemKey"></param>
+        /// <param name="partitionKey">Data Partition key</param>
+        /// <returns>An instance of <see cref="WorkitemInstanceIdentifier"/> representing the <paramref name="dicomDataset"/>.</returns>
+        public static WorkitemInstanceIdentifier ToWorkitemInstanceIdentifier(this DicomDataset dicomDataset, long workitemKey = default, int partitionKey = default)
+        {
+            EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
+
+            // Note: Here we 'GetSingleValueOrDefault' and let the constructor validate the identifier.
+            return new WorkitemInstanceIdentifier(
+                dicomDataset.GetSingleValueOrDefault(DicomTag.SOPInstanceUID, string.Empty),
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Extensions/WorkitemDatasetExtensions.cs:27]: Are we sure about this? I dont think this tag is populated

File: src/Microsoft.Health.Dicom.Api/Extensions/WorkitemResponseStatusExtensions.cs [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+
+using System.Collections.Generic;
+using System.Net;
+using Microsoft.Health.Dicom.Core.Messages.WorkitemMessages;
+
+namespace Microsoft.Health.Dicom.Api.Extensions
+{
+    public static class WorkitemResponseStatusExtensions
+    {
+        private static readonly IReadOnlyDictionary<WorkitemResponseStatus, HttpStatusCode> StoreResponseStatusToHttpStatusCodeMapping =
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Api/Extensions/WorkitemResponseStatusExtensions.cs:14]: Rename Store

File: src/Microsoft.Health.Dicom.Api/Extensions/WorkitemResponseStatusExtensions.cs [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+            new Dictionary<WorkitemResponseStatus, HttpStatusCode>()
+            {
+                { WorkitemResponseStatus.None, HttpStatusCode.NoContent },
+                { WorkitemResponseStatus.Success, HttpStatusCode.OK },
+                { WorkitemResponseStatus.Failure, HttpStatusCode.Conflict },
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Api/Extensions/WorkitemResponseStatusExtensions.cs:19]: This needs to be revisited

File: src/Microsoft.Health.Dicom.Api/Features/Filters/UpsRsFeatureFilterAttribute.cs [hunk 18]
```diff
@@ -0,0 +1,38 @@
...
+            EnsureArg.IsNotNull(featureConfiguration, nameof(featureConfiguration));
+
+            _isUpsRsEnabled = featureConfiguration.Value.EnableUpsRs;
+        }
+
+        public async override Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+        {
+            EnsureArg.IsNotNull(context, nameof(context));
+
+            if (_isUpsRsEnabled)
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Api/Features/Filters/UpsRsFeatureFilterAttribute.cs:30]: if(!_isUpsRsEnabled)

File: src/Microsoft.Health.Dicom.Api/Features/Routing/UrlResolver.cs [hunk 19]
```diff
@@ -96,6 +96,24 @@ public Uri ResolveRetrieveStudyUri(string studyInstanceUid)
             return RouteUri(routeName, routeValues);
         }
 
+        /// <inheritdoc />
+        public Uri ResolveRetrieveWorkitemUri(string workitemInstanceUid)
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Api/Features/Routing/UrlResolver.cs:100]: We need a test

File: src/Microsoft.Health.Dicom.Core/Extensions/DicomMediatorExtensions.cs [hunk 20]
```diff
@@ -216,5 +217,12 @@ public static class DicomMediatorExtensions
             EnsureArg.IsNotNull(mediator, nameof(mediator));
             return mediator.Send(new GetPartitionsRequest(), cancellationToken);
         }
+
+        public static Task<AddWorkitemResponse> AddWorkitemAsync(
+            this IMediator mediator, Stream requestBody, string requestContentType, string upsInstanceUid, CancellationToken cancellationToken)
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Extensions/DicomMediatorExtensions.cs:222]: workitemInstanceUID

File: src/Microsoft.Health.Dicom.Core/Extensions/WorkitemDatasetExtensions.cs [hunk 21]
```diff
@@ -0,0 +1,32 @@
...
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using FellowOakDicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Features.Model;
+
+namespace Microsoft.Health.Dicom.Core.Extensions
+{
+    public static class WorkitemDatasetExtensions
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Extensions/WorkitemDatasetExtensions.cs:12]: we can remove this file

File: src/Microsoft.Health.Dicom.Core/Features/Common/LoggingWorkitemStore.cs [hunk 22]
```diff
@@ -0,0 +1,69 @@
...
+
+namespace Microsoft.Health.Dicom.Core.Features.Common
+{
+    public class LoggingWorkitemStore : IWorkitemStore
+    {
+        private static readonly Action<ILogger, string, Exception> LogAddWorkitemDelegate =
+               LoggerMessage.Define<string>(
+                   LogLevel.Debug,
+                   default,
+                   "Storing DICOM instance workitem file with '{WorkitemInstanceIdentifier}'.");
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Features/Common/LoggingWorkitemStore.cs:22]: Adding

File: src/Microsoft.Health.Dicom.Core/Features/Routing/IUrlResolver.cs [hunk 23]
```diff
@@ -47,5 +47,12 @@ public interface IUrlResolver
         /// <param name="instanceIdentifier">The identifier to the instance.</param>
         /// <returns>An instance of <see cref="Uri"/> pointing to where the instance can be retrieved.</returns>
         Uri ResolveRetrieveInstanceUri(InstanceIdentifier instanceIdentifier);
+
+        /// <summary>
+        ///
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Features/Routing/IUrlResolver.cs:52]: doc?

File: src/Microsoft.Health.Dicom.Core/Features/Workitem/AddWorkitemResponseBuilder.cs [hunk 24]
```diff
@@ -0,0 +1,63 @@
...
+            _urlResolver = urlResolver;
+        }
+
+        /// <inheritdoc />
+        public AddWorkitemResponse BuildResponse()
+        {
+            Uri url = null;
+            WorkitemResponseStatus status = WorkitemResponseStatus.Failure;
+
+            if (_dataset.TryGetSingleValue<string>(DicomTag.RequestedSOPInstanceUID, out var workitemInstanceUid)
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Features/Workitem/AddWorkitemResponseBuilder.cs:36]: AffectedSOPInstanceUID

File: src/Microsoft.Health.Dicom.Core/Features/Workitem/IWorkItemService.cs [hunk 25]
```diff
@@ -0,0 +1,33 @@
...
+
+namespace Microsoft.Health.Dicom.Core.Features.Workitem
+{
+    /// <summary>
+    /// Provides functionality to process the <see cref="DicomDataset"/>.
+    /// </summary>
+    public interface IWorkitemService
+    {
+        /// <summary>
+        /// Asynchronously processes the ...
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Features/Workitem/IWorkItemService.cs:19]: Comment?

File: src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemOrchestrator.cs [hunk 26]
```diff
@@ -0,0 +1,108 @@
...
+            }
+
+            try
+            {
+                await _indexWorkitemStore
+                    .DeleteWorkitemAsync(identifier.PartitionKey, identifier.WorkitemUid, cancellationToken)
+                    .ConfigureAwait(false);
+            }
+            catch
+            {
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemOrchestrator.cs:96]: Add a logger

File: src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemSerializer.cs [hunk 27]
```diff
@@ -0,0 +1,62 @@
...
+
+                return datasets;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Provides functionality to serialize and deserialize workitem instance.
+    /// </summary>
+    public interface IWorkitemSerializer
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemSerializer.cs:52]: Move it to a separate file

File: src/Microsoft.Health.Dicom.Api/Extensions/WorkitemResponseStatusExtensions.cs (hunk 16 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.Api/Extensions/WorkitemResponseStatusExtensions.cs]: Done

File: src/Microsoft.Health.Dicom.Api/Features/Filters/UpsRsFeatureFilterAttribute.cs (hunk 18 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.Api/Features/Filters/UpsRsFeatureFilterAttribute.cs]: Done

File: src/Microsoft.Health.Dicom.Core/Extensions/DicomMediatorExtensions.cs (hunk 20 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.Core/Extensions/DicomMediatorExtensions.cs]: Done

File: src/Microsoft.Health.Dicom.Core/Extensions/WorkitemDatasetExtensions.cs (hunk 21 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.Core/Extensions/WorkitemDatasetExtensions.cs]: Done

File: src/Microsoft.Health.Dicom.Core/Features/Common/LoggingWorkitemStore.cs (hunk 22 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.Core/Features/Common/LoggingWorkitemStore.cs]: Done

File: src/Microsoft.Health.Dicom.Core/Features/Routing/IUrlResolver.cs (hunk 23 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.Core/Features/Routing/IUrlResolver.cs]: Done

File: src/Microsoft.Health.Dicom.Core/Features/Workitem/AddWorkitemResponseBuilder.cs (hunk 24 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.Core/Features/Workitem/AddWorkitemResponseBuilder.cs]: Done

File: src/Microsoft.Health.Dicom.Core/Features/Workitem/IWorkItemService.cs (hunk 25 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.Core/Features/Workitem/IWorkItemService.cs]: Done

File: src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemOrchestrator.cs (hunk 26 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemOrchestrator.cs:104]: Done

File: src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemSerializer.cs (hunk 27 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemSerializer.cs]: Done

File: src/Microsoft.Health.Dicom.Api/Extensions/WorkitemResponseStatusExtensions.cs (hunk 17 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.Api/Extensions/WorkitemResponseStatusExtensions.cs:19]: This requires revisiting the Store handler response codes as well. We should consider addressing this in a separate PR.

File: src/Microsoft.Health.Dicom.Core/Extensions/WorkitemDatasetExtensions.cs (hunk 15 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.Core/Extensions/WorkitemDatasetExtensions.cs]: Done

File: test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/DataPartitionEnabledHttpIntegrationTestFixture.cs (hunk 14 above)
Comment (shivakumar-ms) [reply test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/DataPartitionEnabledHttpIntegrationTestFixture.cs]: Done

**Review by jovinson-ms — APPROVED** [review]

**Review by bcarthic — APPROVED** [review]
